apps/web/src/ui/stake/SushiBarProvider.tsx

```diff
@@ -19,34 +29,68 @@ export const SushiBarProvider: FC<{
   children: ReactNode
   watch?: boolean
 }> = ({ children }) => {
-  const { data, isLoading, isError } = useBarData()
+  const {
+    data,
+    isLoading: isLoadingBarData,
+    isError: isErrorBarData,
+  } = useBarData()
 
-  const [sushiBalance, totalSupply, apy] = useMemo(
-    () => [
-      tryParseAmount(
-        (data?.sushiSupply ?? 0).toString(),
-        SUSHI[ChainId.ETHEREUM],
-      ),
-      tryParseAmount(
-        (data?.xSushiSupply ?? 0).toString(),
-        XSUSHI[ChainId.ETHEREUM],
-      ),
-      data && data?.apr1m !== undefined ? Number(data.apr1m) * 12 : undefined,
+  const {
+    data: balanceAndSupplyData,
+    isLoading: isLoadingSushiBalanceData,
+    isError: isErrorSushiBalanceData,
+    queryKey: balanceAndSupplyQueryKey,
+  } = useReadContracts({
+    allowFailure: false,
+    contracts: [
+      {
+        address: SUSHI_ADDRESS[ChainId.ETHEREUM],
+        abi: erc20Abi,
+        functionName: 'balanceOf',
+        args: [XSUSHI_ADDRESS[ChainId.ETHEREUM]],
+        chainId: ChainId.ETHEREUM,
+      },
+      {
+        address: XSUSHI_ADDRESS[ChainId.ETHEREUM],
+        chainId: ChainId.ETHEREUM,
+        abi: erc20Abi,
+        functionName: 'totalSupply' as const,
+      },
     ],
-    [data],
-  )
+    query: {
+      select(data) {
+        return [
+          Amount.fromRawAmount(SUSHI[ChainId.ETHEREUM], data[0]),
+          Amount.fromRawAmount(XSUSHI[ChainId.ETHEREUM], data[1]),
+        ] as const
+      },
+      staleTime: 30000,
+    },
+  })
+
+  useWatchByInterval({ key: balanceAndSupplyQueryKey, interval: 30000 })
```

Comment:
any reason to use useWatchByInterval instead of setting `refetchInterval` on useReadContracts?

Reply:
No. I've changed it to use `refetchInterval` now.